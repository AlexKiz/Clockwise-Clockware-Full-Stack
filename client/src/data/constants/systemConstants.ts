/* eslint-disable max-len */
export const OPENING_HOURS: string[] = ['09:00', '10:00', '11:00', '12:00', '13:00', '14:00', '15:00', '16:00', '17:00', '18:00'];
export const ACCESS_TOKEN = 'accessToken';

export const VALID = {
	NAME: new RegExp(/^([(A-Za-zА-Яа-я]{3,49})$|^([A-Za-zА-Яа-я]{3,49}[\s]{1}[A-Za-zА-Яа-я]{3,50})$/),
	EMAIL: new RegExp(/^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/),
	CITY: new RegExp(/^[A-Za-zА-Яа-я]{3,100}$|^[A-Za-zА-Яа-я]{3,49}[-\s]{1}[A-Za-zА-Яа-я]{3,50}$/),
	PASSWORD: new RegExp(/^(?=.*[a-z])(?=.*[A-Z])(?=.*\d)(?=.*[@$!%*?&]).{0,}$/),
};

export enum ROLE {
	ADMIN = 'admin',
    MASTER = 'master',
	CLIENT = 'client',
}

export enum SORTED_FIELD {
	ID = 'id',
	NAME = 'name',
	EMAIL = 'email',
	RATING = 'rating',
	CLOCK_SIZE = 'clock.size',
	USER_NAME = 'user.name',
	USER_EMAIL = 'user.email',
	CITY_NAME = 'city.name',
	MASTER_NAME = 'master.name',
	START_WORK_ON = 'startWorkOn',
	END_WORK_ON = 'endWorkOn',
	SMALL_CLOCKS = 'smallClocks',
	MEDIUM_CLOCKS = 'mediumClocks',
	LARGE_CLOCKS = 'largeClocks',
	COMPLETED = 'completed',
	UNCOMPLETED = 'uncompleted',
	EARNED_AMOUNT = 'earnedAmount'
}

export enum SORTING_ORDER {
	ASC = 'asc',
	DESC = 'desc'
}
