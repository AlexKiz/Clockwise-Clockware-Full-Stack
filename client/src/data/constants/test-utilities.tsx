import React from 'react';
import {render} from '@testing-library/react';
import {MemoryRouter, Route} from 'react-router-dom';


export const renderWithRouter = ({component, path, exact = false}) => (
	render(
		<MemoryRouter initialEntries={[path]}>
			<Route path={`${path}`} exact={exact}>
				{component}
			</Route>
		</MemoryRouter>,
	)
);

export const rerenderWrapper = ({component, path, exact = false}) => {
	return (
		<MemoryRouter initialEntries={[path]}>
			<Route path={`${path}`} exact={exact}>
				{component}
			</Route>
		</MemoryRouter>
	);
};

export * from '@testing-library/react';
