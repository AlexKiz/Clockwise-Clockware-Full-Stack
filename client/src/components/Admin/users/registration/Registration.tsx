import axios from 'axios';
import React, {useState, useEffect, FC} from 'react';
import PublicHeader from '../../../Headers/PublicHeader';
import classes from './registration-form.module.css';
import {RegistrationProps, validate} from './componentConstants';
import {useFormik} from 'formik';
import {URL} from '../../../../data/constants/routeConstants';
import {ROLE} from '../../../../data/constants/systemConstants';
import {
	Button,
	Checkbox,
	FilledInput,
	FormControl,
	FormControlLabel,
	FormGroup,
	FormHelperText,
	IconButton,
	InputAdornment,
	InputLabel,
	MenuItem,
	OutlinedInput,
	Select,
	Stack,
	TextField,
	Typography,
	CircularProgress,
} from '@mui/material';
import {Visibility, VisibilityOff} from '@mui/icons-material';
import {AlertNotification, City} from 'src/data/types/types';
import AlertMessage from 'src/components/Notification/AlertMessage';
import {useTranslation} from 'react-i18next';


const Registration:FC<RegistrationProps> = () => {
	const {t} = useTranslation();
	const [cities, setCities] = useState<City[]>([]);

	const [showPassword, setShowPassword] = useState<boolean>(false);
	const [alertOptions, setAlertOptions] = useState<AlertNotification>({
		notify: false,
		type: 'success',
		message: '',
	});
	const [loading, setLoading] = useState<boolean>(false);

	const isOpen = (value: boolean) => {
		setAlertOptions({...alertOptions, notify: value});
	};

	const formik = useFormik({
		initialValues: {
			email: '',
			firstName: '',
			lastName: '',
			password: '',
			checkPassword: '',
			licenseAcception: false,
			isMaster: false,
			citiesId: [] as number[],

		},
		validate,
		onSubmit: async (values) => {
			setLoading(true);
			await axios.post(URL.REGISTRATION,
				{
					email: values.email,
					name: `${values.firstName} ${values.lastName}`,
					password: values.password,
					citiesId: values.citiesId,
					role: values.isMaster ? ROLE.MASTER : ROLE.CLIENT,
				}).then(() => {
				setLoading(false);
				setAlertOptions({
					message: 'Check your email to verify account',
					type: 'success',
					notify: true,
				});
				values.licenseAcception = false,
				values.isMaster = false,
				formik.resetForm();
			}).catch((error) => {
				setLoading(false);
				if (error.response.status === 400) {
					setAlertOptions({
						message: 'User with current email already exists',
						type: 'error',
						notify: true,
					});
					values.email = '';
				} else {
					setAlertOptions({
						message: 'Something went wrong',
						type: 'error',
						notify: true,
					});
				}
			});
		},
	});

	useEffect(() => {
		const readCitiesData = async () => {
			const {data} = await axios.get<City[]>(URL.CITY);

			if (data.length) {
				setCities(data);
			}
		};

		readCitiesData();
	}, []);


	return (
		<div>
			<PublicHeader/>
			<div className={classes.container}>
				<div className={classes.container_form}>
					<form className={classes.form} onSubmit={formik.handleSubmit}>
						<Stack direction='column' justifyContent='center' spacing={1}>
							<div className={classes.form_section}>
								<div className={classes.form_input__label}>
									<Typography
										variant="h6"
										gutterBottom
										component="label"
									>
										{t('registration.email')}
									</Typography>
								</div>
								<TextField
									id="email"
									name='email'
									label={t('labels.email')}
									placeholder="example@mail.com"
									variant="filled"
									size="small"
									margin="dense"
									fullWidth
									value={formik.values.email}
									onChange={formik.handleChange}
									onBlur={formik.handleBlur}
									error={formik.touched.email && Boolean(formik.errors.email)}
									helperText={formik.touched.email && formik.errors.email}
									required
									inputProps={{
										'data-testid': 'registration-email-input',
									}}
								/>
							</div>
							<div className={classes.form_section}>
								<div className={classes.form_input__label}>
									<Typography
										variant="h6"
										gutterBottom
										component="label"
									>
										{t('registration.password')}
									</Typography>
								</div>
								<FormControl fullWidth variant="filled">
									<InputLabel>{t('labels.password')}</InputLabel>
									<FilledInput
										id="filled-adornment-password"
										type={showPassword ? 'text' : 'password'}
										value={formik.values.password}
										onChange={formik.handleChange('password')}
										onBlur={formik.handleBlur('password')}
										error={formik.touched.password && Boolean(formik.errors.password)}
										required
										endAdornment={
											<InputAdornment position="end">
												<IconButton
													aria-label="toggle password visibility"
													onClick={() => setShowPassword((showPassword) => !showPassword)}
													edge="end"
												>
													{showPassword ? <VisibilityOff /> : <Visibility />}
												</IconButton>
											</InputAdornment>
										}
										inputProps={{
											'data-testid': 'registration-password-input',
										}}
									/>
									<FormHelperText error> {formik.touched.password && formik.errors.password} </FormHelperText>
								</FormControl>
								<FormControl fullWidth variant="filled">
									<InputLabel>{t('labels.confirmPassword')}</InputLabel>
									<FilledInput
										id="filled-adornment-password"
										type={showPassword ? 'text' : 'password'}
										value={formik.values.checkPassword}
										onChange={formik.handleChange('checkPassword')}
										onBlur={formik.handleBlur('checkPassword')}
										error={formik.touched.checkPassword && Boolean(formik.errors.checkPassword)}
										required
										endAdornment={
											<InputAdornment position="end">
												<IconButton
													aria-label="toggle password visibility"
													onClick={() => setShowPassword((showPassword) => !showPassword)}
													edge="end"
												>
													{showPassword ? <VisibilityOff /> : <Visibility />}
												</IconButton>
											</InputAdornment>
										}
										inputProps={{
											'data-testid': 'registration-repeatPassword-input',
										}}
									/>
									<FormHelperText error> {formik.touched.checkPassword && formik.errors.checkPassword} </FormHelperText>
								</FormControl>
							</div>
							<div className={classes.from_section}>
								<div className={classes.form_input__label}>
									<Typography
										variant="h6"
										gutterBottom
										component="label"
									>
										{t('registration.name')}
									</Typography>
								</div>
								<TextField
									id="firstName"
									name='firstName'
									label={t('labels.firstName')}
									placeholder="First Name"
									variant="filled"
									size="small"
									margin="dense"
									fullWidth
									value={formik.values.firstName}
									onChange={formik.handleChange}
									onBlur={formik.handleBlur}
									error={formik.touched.firstName && Boolean(formik.errors.firstName)}
									helperText={formik.touched.firstName && formik.errors.firstName}
									required
									inputProps={{
										'data-testid': 'registration-firstName-input',
									}}
								/>
								<TextField
									id="lastName"
									name='lastName'
									label={t('labels.lastName')}
									placeholder="Last Name"
									variant="filled"
									size="small"
									margin="dense"
									fullWidth
									value={formik.values.lastName}
									onChange={formik.handleChange}
									onBlur={formik.handleBlur}
									error={formik.touched.lastName && Boolean(formik.errors.lastName)}
									helperText={formik.touched.lastName && formik.errors.lastName}
									required
									inputProps={{
										'data-testid': 'registration-lastName-input',
									}}
								/>
							</div>
							<div className={classes.form_section}>
								<FormControl
									required
									error={formik.touched.licenseAcception && Boolean(formik.errors.licenseAcception)}
									component="fieldset"
									sx={{m: 3, width: '100%'}}
									variant="outlined"
								>
									<FormGroup row>
										<FormControlLabel
											control={
												<Checkbox
													onChange={formik.handleChange('licenseAcception')}
													checked={formik.values.licenseAcception}
													required
													inputProps={{
														// @ts-ignore
														'data-testid': 'registration-license-checkbox',
													}}
												/>
											}
											label={String(t('labels.licenseTerm'))}
										/>
										<FormControlLabel
											control={
												<Checkbox
													onChange={formik.handleChange('isMaster')}
													checked={formik.values.isMaster}
													value=''
													inputProps={{
														// @ts-ignore
														'data-testid': 'registration-isMaster-checkbox',
													}}
												/>
											}
											label={String(t('labels.asMaster'))}
										/>
									</FormGroup>
									<FormHelperText error>
										{formik.touched.licenseAcception && Boolean(formik.errors.licenseAcception)}
									</FormHelperText>
								</FormControl>
							</div>

							{ formik.values.isMaster &&
								<div className={classes.form_section}>
									<div className={classes.form_input__label}>
										<Typography
											variant="h6"
											gutterBottom
											component="label"
										>
											{t('registration.city')}
										</Typography>
									</div>
									<FormControl
										fullWidth
										error={formik.touched.citiesId && Boolean(formik.errors.citiesId)}
									>
										<InputLabel id="cities">{t('labels.cities')}</InputLabel>
										<Select
											id='citiesId'
											name='citiesId'
											labelId='cities'
											placeholder='City'
											displayEmpty
											multiple
											onChange={formik.handleChange}
											value={formik.values.citiesId}
											onBlur={formik.handleBlur}
											input={<OutlinedInput label="Cities" data-testid='registration-cities-select'/>}
											MenuProps={{
												sx: {
													'&& .Mui-selected': {
														backgroundColor: '#0094fd5c',
													},
												},
											}}
										>
											{cities.map((city) => (
												<MenuItem
													key={city.id}
													value={city.id}
												>
													{city.name}
												</MenuItem>
											))}
										</Select>
										<FormHelperText> {formik.touched.citiesId && formik.errors.citiesId} </FormHelperText>
									</FormControl>
								</div>
							}
							<div className={classes.form_section}>
								<Button
									disabled={!(formik.isValid && formik.dirty && !loading)}
									color='success'
									variant="contained"
									type="submit"
									className={classes.form_btn}
									style={ {fontSize: 18, borderRadius: 15} }
								>
									{t('buttons.submit')}
									{loading && <CircularProgress
										size={56}
										color="success"
										sx={{
											position: 'absolute',
											top: '50%',
											left: '50%',
											marginTop: '-28px',
											marginLeft: '-28px',
										}}
									/>}
								</Button>
							</div>
						</Stack>
					</form>
					{
						alertOptions.notify &&
						<AlertMessage
							alertType={alertOptions.type}
							message={alertOptions.message}
							isOpen={isOpen}
							notify={alertOptions.notify}
						/>
					}
				</div>
			</div>
		</div>
	);
};

export default Registration;
