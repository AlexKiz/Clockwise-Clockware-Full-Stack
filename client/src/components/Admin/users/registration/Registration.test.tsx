import React from 'react';
import Registration from './Registration';
import {act, cleanup, fireEvent} from '@testing-library/react';
import {renderWithRouter, rerenderWrapper} from '../../../../data/constants/test-utilities';

describe('Registration Form Component', () => {
	beforeEach(() => {
		cleanup();
	});
	jest.mock('react-router-dom', () => ({
		...jest.requireActual('react-router-dom'),
		useHistory: () => ({
			push: jest.fn(),
		}),
		useLocation: () => ({
			pathname: '/registration-form',
		}),
	}));

	it('matches snapshot', () => {
		const {asFragment} = renderWithRouter({
			component: <Registration />,
			path: '/registration-form',
		});

		expect(asFragment()).toMatchSnapshot();
	});

	it('shows inputs with empty values', () => {
		const {getByTestId} = renderWithRouter({
			component: <Registration />,
			path: '/registration-form',
		});

		expect((getByTestId('registration-email-input') as HTMLInputElement).value).toBe('');
		expect((getByTestId('registration-password-input') as HTMLInputElement).value).toBe('');
		expect((getByTestId('registration-repeatPassword-input') as HTMLInputElement).value).toBe('');
		expect((getByTestId('registration-firstName-input') as HTMLInputElement).value).toBe('');
		expect((getByTestId('registration-lastName-input') as HTMLInputElement).value).toBe('');
		expect((getByTestId('registration-license-checkbox') as HTMLInputElement).checked).toEqual(false);
		expect((getByTestId('registration-isMaster-checkbox') as HTMLInputElement).checked).toEqual(false);
	});

	it('triggers event handlers on UI elements', () => {
		const {getByTestId, rerender} = renderWithRouter({
			component: <Registration />,
			path: '/registration-form',
		});

		act(() => {
			fireEvent.change(getByTestId('registration-email-input'), {
				target: {value: 'testRegistration@gmail.com'},
			});
			fireEvent.change(getByTestId('registration-password-input'), {
				target: {value: 'TestPassword1!'},
			});
			fireEvent.change(getByTestId('registration-repeatPassword-input'), {
				target: {value: 'TestPassword1!'},
			});
			fireEvent.change(getByTestId('registration-firstName-input'), {
				target: {value: 'First'},
			});
			fireEvent.change(getByTestId('registration-lastName-input'), {
				target: {value: 'Last'},
			});
			fireEvent.click(getByTestId('registration-license-checkbox'));
			fireEvent.click(getByTestId('registration-isMaster-checkbox'));
		});

		rerender(rerenderWrapper({
			component: <Registration />,
			path: '/registration-form',
		}));

		expect((getByTestId('registration-email-input') as HTMLInputElement).value).toBe('testRegistration@gmail.com');
		expect((getByTestId('registration-password-input') as HTMLInputElement).value).toBe('TestPassword1!');
		expect((getByTestId('registration-repeatPassword-input') as HTMLInputElement).value).toBe('TestPassword1!');
		expect((getByTestId('registration-firstName-input') as HTMLInputElement).value).toBe('First');
		expect((getByTestId('registration-lastName-input') as HTMLInputElement).value).toBe('Last');
		expect((getByTestId('registration-license-checkbox') as HTMLInputElement).checked).toEqual(true);
		expect((getByTestId('registration-isMaster-checkbox') as HTMLInputElement).checked).toEqual(true);
		expect((getByTestId('registration-cities-select') as HTMLInputElement)).toBeInTheDocument();
	});
});
