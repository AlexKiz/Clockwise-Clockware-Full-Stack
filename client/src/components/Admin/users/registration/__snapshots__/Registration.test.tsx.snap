// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Registration Form Component matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <header>
      <div
        class="wrapper_header"
      >
        <div
          class="wrapper_logo"
        >
          <a
            href="/"
          >
            <div
              class="inner_logo"
            >
              <div
                class="inner_logo_img"
              >
                <div
                  class="logo_img1a"
                >
                  <div
                    class="logo_img1b"
                  >
                    <div
                      class="logo_img1c"
                    />
                  </div>
                </div>
              </div>
              <div
                class="logo_text1d"
              />
            </div>
          </a>
        </div>
        <nav>
          <ul
            class="nav__links"
          >
            <li
              class=""
            >
              <a
                href="/"
              >
                Make order
              </a>
            </li>
            <li
              class=""
            >
              <a
                href="/login"
              >
                Log in
              </a>
            </li>
            <li
              class="active"
            >
              <a
                href="/registration-form"
              >
                Sign up
              </a>
            </li>
            <li
              class=""
            >
              <a
                href="/blog"
              >
                Blog
              </a>
            </li>
            <li>
              <div
                class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary  css-1s4qybw-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
              >
                <div
                  aria-expanded="false"
                  aria-haspopup="listbox"
                  class="MuiSelect-select MuiSelect-outlined MuiOutlinedInput-input MuiInputBase-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
                  role="button"
                  tabindex="0"
                >
                  English
                </div>
                <input
                  aria-hidden="true"
                  class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
                  tabindex="-1"
                  value="en"
                />
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
                  data-testid="ArrowDropDownIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M7 10l5 5 5-5z"
                  />
                </svg>
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-hdw1oc"
                  >
                    <span
                      class="notranslate"
                    >
                      ​
                    </span>
                  </legend>
                </fieldset>
              </div>
            </li>
          </ul>
        </nav>
      </div>
    </header>
    <div
      class="container"
    >
      <div
        class="container_form"
      >
        <form
          class="form"
        >
          <div
            class="css-1nw2civ-MuiStack-root"
          >
            <div
              class="form_section"
            >
              <div
                class="form_input__label"
              >
                <label
                  class="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom css-18k87ye-MuiTypography-root"
                >
                  Enter Email:
                </label>
              </div>
              <div
                class="MuiFormControl-root MuiFormControl-marginDense MuiFormControl-fullWidth MuiTextField-root css-1z10yd4-MuiFormControl-root-MuiTextField-root"
              >
                <label
                  class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-filled MuiFormLabel-root MuiFormLabel-colorPrimary Mui-required css-1mog9k0-MuiFormLabel-root-MuiInputLabel-root"
                  data-shrink="false"
                  for="email"
                  id="email-label"
                >
                  Email
                  <span
                    aria-hidden="true"
                    class="MuiInputLabel-asterisk MuiFormLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
                  >
                     *
                  </span>
                </label>
                <div
                  class="MuiFilledInput-root MuiFilledInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-sizeSmall css-10i04qz-MuiInputBase-root-MuiFilledInput-root"
                >
                  <input
                    aria-invalid="false"
                    class="MuiFilledInput-input MuiInputBase-input MuiInputBase-inputSizeSmall css-c5v1qu-MuiInputBase-input-MuiFilledInput-input"
                    data-testid="registration-email-input"
                    id="email"
                    name="email"
                    placeholder="example@mail.com"
                    required=""
                    type="text"
                    value=""
                  />
                </div>
              </div>
            </div>
            <div
              class="form_section"
            >
              <div
                class="form_input__label"
              >
                <label
                  class="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom css-18k87ye-MuiTypography-root"
                >
                  Enter Password:
                </label>
              </div>
              <div
                class="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
              >
                <label
                  class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-filled MuiFormLabel-root MuiFormLabel-colorPrimary css-e4w4as-MuiFormLabel-root-MuiInputLabel-root"
                  data-shrink="false"
                >
                  Password
                </label>
                <div
                  class="MuiFilledInput-root MuiFilledInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedEnd css-f6v4u4-MuiInputBase-root-MuiFilledInput-root"
                  required=""
                >
                  <input
                    aria-invalid="false"
                    class="MuiFilledInput-input MuiInputBase-input MuiInputBase-inputAdornedEnd css-1gctnaj-MuiInputBase-input-MuiFilledInput-input"
                    data-testid="registration-password-input"
                    id="filled-adornment-password"
                    required=""
                    type="password"
                    value=""
                  />
                  <div
                    class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-filled MuiInputAdornment-sizeMedium css-jv61v3-MuiInputAdornment-root"
                  >
                    <button
                      aria-label="toggle password visibility"
                      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd MuiIconButton-sizeMedium css-1yq5fb3-MuiButtonBase-root-MuiIconButton-root"
                      tabindex="0"
                      type="button"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                        data-testid="VisibilityIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M12 4.5C7 4.5 2.73 7.61 1 12c1.73 4.39 6 7.5 11 7.5s9.27-3.11 11-7.5c-1.73-4.39-6-7.5-11-7.5zM12 17c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5zm0-8c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3z"
                        />
                      </svg>
                      <span
                        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                      />
                    </button>
                  </div>
                </div>
                <p
                  class="MuiFormHelperText-root Mui-error MuiFormHelperText-sizeMedium MuiFormHelperText-contained css-1wc848c-MuiFormHelperText-root"
                >
                    
                </p>
              </div>
              <div
                class="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
              >
                <label
                  class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-filled MuiFormLabel-root MuiFormLabel-colorPrimary css-e4w4as-MuiFormLabel-root-MuiInputLabel-root"
                  data-shrink="false"
                >
                  Confirm password
                </label>
                <div
                  class="MuiFilledInput-root MuiFilledInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedEnd css-f6v4u4-MuiInputBase-root-MuiFilledInput-root"
                  required=""
                >
                  <input
                    aria-invalid="false"
                    class="MuiFilledInput-input MuiInputBase-input MuiInputBase-inputAdornedEnd css-1gctnaj-MuiInputBase-input-MuiFilledInput-input"
                    data-testid="registration-repeatPassword-input"
                    id="filled-adornment-password"
                    required=""
                    type="password"
                    value=""
                  />
                  <div
                    class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-filled MuiInputAdornment-sizeMedium css-jv61v3-MuiInputAdornment-root"
                  >
                    <button
                      aria-label="toggle password visibility"
                      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd MuiIconButton-sizeMedium css-1yq5fb3-MuiButtonBase-root-MuiIconButton-root"
                      tabindex="0"
                      type="button"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                        data-testid="VisibilityIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M12 4.5C7 4.5 2.73 7.61 1 12c1.73 4.39 6 7.5 11 7.5s9.27-3.11 11-7.5c-1.73-4.39-6-7.5-11-7.5zM12 17c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5zm0-8c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3z"
                        />
                      </svg>
                      <span
                        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                      />
                    </button>
                  </div>
                </div>
                <p
                  class="MuiFormHelperText-root Mui-error MuiFormHelperText-sizeMedium MuiFormHelperText-contained css-1wc848c-MuiFormHelperText-root"
                >
                    
                </p>
              </div>
            </div>
            <div
              class="from_section"
            >
              <div
                class="form_input__label"
              >
                <label
                  class="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom css-18k87ye-MuiTypography-root"
                >
                  Enter your name:
                </label>
              </div>
              <div
                class="MuiFormControl-root MuiFormControl-marginDense MuiFormControl-fullWidth MuiTextField-root css-1z10yd4-MuiFormControl-root-MuiTextField-root"
              >
                <label
                  class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-filled MuiFormLabel-root MuiFormLabel-colorPrimary Mui-required css-1mog9k0-MuiFormLabel-root-MuiInputLabel-root"
                  data-shrink="false"
                  for="firstName"
                  id="firstName-label"
                >
                  First Name
                  <span
                    aria-hidden="true"
                    class="MuiInputLabel-asterisk MuiFormLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
                  >
                     *
                  </span>
                </label>
                <div
                  class="MuiFilledInput-root MuiFilledInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-sizeSmall css-10i04qz-MuiInputBase-root-MuiFilledInput-root"
                >
                  <input
                    aria-invalid="false"
                    class="MuiFilledInput-input MuiInputBase-input MuiInputBase-inputSizeSmall css-c5v1qu-MuiInputBase-input-MuiFilledInput-input"
                    data-testid="registration-firstName-input"
                    id="firstName"
                    name="firstName"
                    placeholder="First Name"
                    required=""
                    type="text"
                    value=""
                  />
                </div>
              </div>
              <div
                class="MuiFormControl-root MuiFormControl-marginDense MuiFormControl-fullWidth MuiTextField-root css-1z10yd4-MuiFormControl-root-MuiTextField-root"
              >
                <label
                  class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-filled MuiFormLabel-root MuiFormLabel-colorPrimary Mui-required css-1mog9k0-MuiFormLabel-root-MuiInputLabel-root"
                  data-shrink="false"
                  for="lastName"
                  id="lastName-label"
                >
                  Last Name
                  <span
                    aria-hidden="true"
                    class="MuiInputLabel-asterisk MuiFormLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
                  >
                     *
                  </span>
                </label>
                <div
                  class="MuiFilledInput-root MuiFilledInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-sizeSmall css-10i04qz-MuiInputBase-root-MuiFilledInput-root"
                >
                  <input
                    aria-invalid="false"
                    class="MuiFilledInput-input MuiInputBase-input MuiInputBase-inputSizeSmall css-c5v1qu-MuiInputBase-input-MuiFilledInput-input"
                    data-testid="registration-lastName-input"
                    id="lastName"
                    name="lastName"
                    placeholder="Last Name"
                    required=""
                    type="text"
                    value=""
                  />
                </div>
              </div>
            </div>
            <div
              class="form_section"
            >
              <fieldset
                class="MuiFormControl-root css-jm0nt1-MuiFormControl-root"
              >
                <div
                  class="MuiFormGroup-root MuiFormGroup-row css-qfz70r-MuiFormGroup-root"
                >
                  <label
                    class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
                  >
                    <span
                      class="MuiCheckbox-root MuiCheckbox-colorPrimary MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
                    >
                      <input
                        class="PrivateSwitchBase-input css-1m9pwf3"
                        data-indeterminate="false"
                        data-testid="registration-license-checkbox"
                        required=""
                        type="checkbox"
                      />
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                        data-testid="CheckBoxOutlineBlankIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                        />
                      </svg>
                      <span
                        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                      />
                    </span>
                    <span
                      class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
                    >
                      *Accept the license terms
                    </span>
                  </label>
                  <label
                    class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
                  >
                    <span
                      class="MuiCheckbox-root MuiCheckbox-colorPrimary MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
                    >
                      <input
                        class="PrivateSwitchBase-input css-1m9pwf3"
                        data-indeterminate="false"
                        data-testid="registration-isMaster-checkbox"
                        type="checkbox"
                        value=""
                      />
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                        data-testid="CheckBoxOutlineBlankIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                        />
                      </svg>
                      <span
                        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                      />
                    </span>
                    <span
                      class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
                    >
                      Sign up as master
                    </span>
                  </label>
                </div>
                <p
                  class="MuiFormHelperText-root Mui-error MuiFormHelperText-sizeMedium MuiFormHelperText-contained Mui-required css-1wc848c-MuiFormHelperText-root"
                />
              </fieldset>
            </div>
            <div
              class="form_section"
            >
              <button
                class="MuiButton-root MuiButton-contained MuiButton-containedSuccess MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root Mui-disabled form_btn css-ke5b6m-MuiButtonBase-root-MuiButton-root"
                disabled=""
                style="font-size: 18px; border-radius: 15px;"
                tabindex="-1"
                type="submit"
              >
                Submit
              </button>
            </div>
          </div>
        </form>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
